Adding event for showing the item sheet on clicking the name or icon

AskMP opened the pull request:
Adding an event listener to show the sheet of the item when clicking on it’s name list item. This includes the icon as it is within the name cell.

AskMP committed:
Adding base event listener

jopeek commented:
Does this interfere with the expansion to see the item details or no?

I’ll look at all these pull requests either later today or tomorrow. Thanks for these!

AskMP committed:
Updated this to the edit button

AskMP committed:
Adding the item name for viewers

Adding the item-name back to clickable to allow viewers to see the description and details. Item sheets are not editable by viewers so no worries there.

AskMP commented:
I removed the ther inline editing of value / denomination in favour of this. The "edit" button should be the manner in which to change those. As for your question about the expansion to see the item details, it shouldn't??? but can you direct me to that? This uses the built-in `item.sheet.render(true);` method if the item exists and the function exists so I’m not doing anything additional or creating a new popup. This is more that I saw there was no functionality attached to the button(s) so I added it.

jopeek commented:
If it's attached to the buttons, it should be fine.

On Sat, Aug 24, 2024 at 10:50 AM Matthew Potter ***@***.***>
wrote:

> I removed the ther inline editing of value / denomination in favour of
> this. The "edit" button should be the manner in which to change those. As
> for your question about the expansion to see the item details, it
> shouldn't??? but can you direct me to that? This uses the built-in
> item.sheet.render(true); method if the item exists and the function
> exists so I’m not doing anything additional or creating a new popup. This
> is more that I saw there was no functionality attached to the button(s) so
> I added it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jopeek/fvtt-loot-sheet-npc-5e/pull/422#issuecomment-2308471393>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJBV6XJ7LSSTB26AWQULIDZTDBUXAVCNFSM6AAAAABNBXDVOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBYGQ3TCMZZGM>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan Ole Peek
http://www.janolepeek.com/

AskMP commented:
Well… the one is however I’ve also added the same click event to the name cell (icon and name) for players to be able to see the details of an item. I noted that the item is NOT editable as a player and there is often a lot more information like description and details of an item in there which was not viewable in the current version. If you would rather an "eye" icon in the far right side of a row to see the details, that may make more sense for accessibility purposes but I know people would likely instinctively click on the name for more information about it.

jopeek commented:
That's what I mean. Clicking the name in the latest version should just
expand the cell to show the details. I'll check these out tomorrow most
likely. Have a game today, so I can't mess with it now ;)

On Sat, Aug 24, 2024 at 10:54 AM Matthew Potter ***@***.***>
wrote:

> Well… the one is however I’ve also added the same click event to the name
> cell (icon and name) for players to be able to see the details of an item.
> I noted that the item is NOT editable as a player and there is often a lot
> more information like description and details of an item in there which was
> not viewable in the current version. If you would rather an "eye" icon in
> the far right side of a row to see the details, that may make more sense
> for accessibility purposes but I know people would likely instinctively
> click on the name for more information about it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jopeek/fvtt-loot-sheet-npc-5e/pull/422#issuecomment-2308472840>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJBV6XYJPY66MGANO7O3OTZTDCDTAVCNFSM6AAAAABNBXDVOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBYGQ3TEOBUGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan Ole Peek
http://www.janolepeek.com/

AskMP commented:
Awesome. Hey… while I have you. Thank you for building this. I do appreciate it. Have a great day! May you be never burdened by falling rocks.

jopeek commented:
Haha, thank you :)

On Sat, Aug 24, 2024 at 10:56 AM Matthew Potter ***@***.***>
wrote:

> Awesome. Hey… while I have you. Thank you for building this. I do
> appreciate it. Have a great day! May you be never burdened by falling rocks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jopeek/fvtt-loot-sheet-npc-5e/pull/422#issuecomment-2308473815>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJBV6TP2ESJCJMKSPKP5ZLZTDCODAVCNFSM6AAAAABNBXDVOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBYGQ3TGOBRGU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Jan Ole Peek
http://www.janolepeek.com/

AskMP commented:
Not needed. There is a different version that SHOULD have been pushed by Foundry. Not getting the updated resulted in an update bug. Removing and reinstalling fixes the issue if you had a version that does not allow for editing and removing items from a lootsheet.
